eqcorrscan/utils/correlate.py

```diff
@@ -582,18 +463,23 @@ def fftw_normxcorr(templates, stream, pads, threaded=False, *args, **kwargs):
         np.ctypeslib.ndpointer(dtype=np.intc,
                                flags=native_str('C_CONTIGUOUS'))]
     restype = ctypes.c_int
-    if threaded and False:  # Forcing non-threaded for now...
+
+    # TODO: convert threaded routine to do post processing in C too - forcing non-threaded for now...
```

Comment:
I think your stuff should be calling the multi_normxcorr_fftw function - these two just call the inner, single channel normxcorr_fftw(_threaded) functions.  These are here to cope with python looping over channels (either using multiprocessing, multithreading or in serial).  The concurrent option should direct to multi_normxcorr_fftw.

Reply:
It's a bit confusing now. The OpenMP one calls `multi_normxcorr_fftw` which then calls `normxcorr_fftw_main` at each iteration of the loop. The multiprocessing/threading/serial version calls `normxcorr_fftw` when threading is set to `False` - this ends up calling `normxcorr_fftw_main` too. However the threaded version calls `normxcorr_fftw_threaded`.

When I made the change to do the sum in place I had to move the post processing code from both python functions into `normxcorr_fftw_main` in C.  I need to replicate this for `normxcorr_fftw_threaded` too otherwise the post processing won't get done in that case - this shouldn't take long.

Reply:
Ah, okay, cool, sorry, I didn't see that!